rdr_service/genomic/genomic_job_components.py

```diff
@@ -302,131 +302,120 @@ def get_aw1_manifest_column_mappings():
+            #  Until verified that this issue has been fixed in manifes
+            # Fix for invalid parent sample values
+            # try:
+            #     parent_sample_id = int(row_copy['parentsampleid'])
+            # except ValueError:
+            #     parent_sample_id = 0
+
+            # Skip rows if biobank_id is an empty string (row is empty well)
+            if row_copy['biobankid'] == "":
+                continue
 
-                    if member is None:
-                        # Check if Programmatic control
-                        # Fix for invalid parent sample values
-                        try:
-                            parent_sample_id = int(row_copy['parentsampleid'])
+            # Find the existing GenomicSetMember
+            # Set the member based on collection tube ID
+            # row_copy['testname'] is the genome type (i.e. aou_array, aou_wgs)
+            member = self.member_dao.get_member_from_collection_tube(row_copy['collectiontubeid'],
+                                                                     row_copy['testname'])
 
-                        except ValueError:
-                            parent_sample_id = 0
+            # If member not found, check if:
+            #  this is a control sample
+            #  or if collection tube was changed by Biobank
+            if member is None:
+                # Check if this is a new control sample and make new member from it
+                if self._check_if_control_sample(int(row_copy['parentsampleid'])) is not None:
+                    logging.warning(f"Control sample found: {row_copy['parentsampleid']}")
 
-                        if self._check_if_control_sample(parent_sample_id) is not None:
-                            logging.warning(f'Control sample found: {parent_sample_id}')
-                            # Ignoring control samples for now
-                            # RDR may need to do something with them in the future
+                    member = self.create_new_member_from_aw1_control_sample(row_copy)
 
-                        else:
-                            logging.error(f"Skipping collection tube ID: {collection_tube_id}, "
-                                          f"genome type: {genome_type}")
+                    # Increment record count for manifest file
+                    count += 1
 
-                        continue
+                    # Update member for PDR
+                    bq_genomic_set_member_update(member.id, project_id=self.controller.bq_project_id)
+                    genomic_set_member_update(member.id)
 
-                # Skip already processed members if failure mode isn't defined in manifest
-                if member.reconcileGCManifestJobRunId is not None and row_copy['failuremode'] in (None, ''):
                     continue
 
-                member.gcSiteId = _site
+                else:
+                    # Not a control sample, continue with next check
+                    pass
```

Comment:
Do you think it could help readability to remove the `continue` above and have the rest of the iteration within the `else` block?

Reply:
I originally had it the way you suggest, but I found that having the `continue` helped make it more readable since we want to terminate the iteration here explicitly. Nesting the remainder of the iteration into the `else` here added more complexity because of the condition on line 343. However, I can nest the block from 360-384 into this condition and that should increase readability.

Reply:
I see what you mean. I'd missed that there was a branch that could have skipped over all of it.